File: Server/src/main/java/org/example/entities/ProductImageEntity.java
package org.example.entities;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "tbl_product_images")
public class ProductImageEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String filename;

    private int priority;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private ProductEntity product;
}

File: Server/src/main/java/org/example/entities/ProductImageEntity.kt
package org.example.entities

import jakarta.persistence.*

@Entity
@Table(name = "product_image")
class ProductImageEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    lateinit var filename: String

    var priority: Int = 0

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    lateinit var product: ProductEntity
}
